str/associatr/get_cis_numpy_files.py

```diff
@@ -31,6 +32,45 @@
 from cpg_utils.hail_batch import get_batch, image_path, init_batch, output_path
 
 
+def extract_genotypes(vcf_file, loci):
+    """
+    Helper function to extract genotypes (SNPs) from a VCF file; target loci specified as a list (can be single or multiple)
+
+    """
+    # Read the VCF file
+    vcf_reader = VCF(vcf_file)
+
+    # Initialize a dictionary to store the results
+    results = {'sample_id': vcf_reader.samples}
+
+    # Initialize results dictionary with empty lists for each locus
+    for locus in loci:
+        results[locus] = ["." for _ in vcf_reader.samples]
+
+    # Iterate through the records in the VCF file for each locus
+    for locus in loci:
+        chrom, pos = locus.split(':')
+        pos = int(pos)
+
+        for record in vcf_reader(f'{chrom}:{pos}-{pos}'):
+            if record.CHROM == chrom and record.POS == pos:
+                for idx, sample in enumerate(record.genotypes):
+                    genotype_call = sample[0] + sample[1]
+                    if genotype_call == 0:
+                        genotype = "0"  # HOM REF
+                    elif genotype_call == 1:
+                        genotype = "1"  # HET
+                    elif genotype_call == 2:
+                        genotype = "2"  # HOM ALT
+                    else:
+                        genotype = "."
```

Comment:
The cyvcf2 representation already has a cleanly parsed way of getting this information 
 as np arrays: https://github.com/brentp/cyvcf2?tab=readme-ov-file#example
```
[variant.]gt_types is array of 0,1,2,3==HOM_REF, HET, UNKNOWN, HOM_ALT
```